Free Ruler/AppDelegate.swift

```diff
@@ -1,57 +1,61 @@
 import Cocoa
-import SwiftyUserDefaults
 
 let env = ProcessInfo.processInfo.environment
 let APP_ICON_HELPER = env["APP_ICON_HELPER"] != nil
 
-extension DefaultsKeys {
-    static let groupedRulers = DefaultsKey<Bool>("groupedRulers", defaultValue: false)
-}
-
 @NSApplicationMain
-class AppDelegate: NSObject, NSApplicationDelegate {
+class AppDelegate: NSObject, NSApplicationDelegate, PreferenceSubscriber {
     
-    let horizontal = RulerController(ruler: Ruler(.horizontal, name: "horizontal-ruler"))
-    let vertical = RulerController(ruler: Ruler(.vertical, name: "vertical-ruler"))
+    var rulers: [RulerController] = []
 
     var timer: Timer?
     let foregroundTimerInterval: TimeInterval = 1 / 60 // 60 fps
     let backgroundTimerInterval: TimeInterval = 1 / 15 // 15 fps
-
+    
     @IBOutlet weak var groupedMenuItem: NSMenuItem!
 
     // MARK: - Lifecycle
     
     func applicationDidFinishLaunching(_ aNotification: Notification) {
-        updateGroupedRulers()
+        updateGroupRulersMenuItem()
+        Prefs.groupRulers.subscribe(self)
         
         if APP_ICON_HELPER {
             let helper = AppIconHelper()
             helper.show()
         } else {
             showRulers()
         }
-        
     }
     
     func showRulers() {
-        horizontal.otherWindow = vertical.rulerWindow
-        vertical.otherWindow = horizontal.rulerWindow
+        rulers = [
+            RulerController(ruler: Ruler(.horizontal, name: "horizontal-ruler")),
+            RulerController(ruler: Ruler(.vertical, name: "vertical-ruler")),
+        ]
```

Comment:
oh yeah I changed App's ruler references to an array [RulerController] instead of hard-coding horizontal and vertical.